app/src/main/java/ryey/easer/skills/usource/do_not_disturb/DoNotDisturbUSourceData.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2016 - 2022 Rui Zhao <renyuneyun@gmail.com>
+ *
+ * This file is part of Easer.
+ *
+ * Easer is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Easer is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Easer.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package ryey.easer.skills.usource.do_not_disturb;
+
+import android.app.NotificationManager;
+import android.os.Build;
+import android.os.Parcel;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.RequiresApi;
+
+import com.orhanobut.logger.Logger;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Arrays;
+
+import ryey.easer.commons.local_skill.IllegalStorageDataException;
+import ryey.easer.commons.local_skill.dynamics.Dynamics;
+import ryey.easer.commons.local_skill.usource.USourceData;
+import ryey.easer.plugin.PluginDataFormat;
+
+@RequiresApi(api = Build.VERSION_CODES.M)
+public class DoNotDisturbUSourceData implements USourceData {
+
+    static int[] doNotDisturbIDs = {
+            NotificationManager.INTERRUPTION_FILTER_UNKNOWN,
+            NotificationManager.INTERRUPTION_FILTER_ALL,
+            NotificationManager.INTERRUPTION_FILTER_PRIORITY,
+            NotificationManager.INTERRUPTION_FILTER_NONE,
+            NotificationManager.INTERRUPTION_FILTER_ALARMS,
+    };
+    boolean[] doNotDisturbModes = new boolean[doNotDisturbIDs.length];
```

Comment:
A better way is to have a `Set` containing only the expected modes (i.e. those are true). Serialization just need to store these; comparison just need to check if that mode is in the set or not; UI just pass the checked ones.

It can also be easily changed to a more human-friendly (for developers and pro-users who may try to read the stored data) version: give a list of names (strings) of each corresponding value; during serialization, change the value to the corresponding names; do the opposite when parsing.

Reply:
The Set is implemented in the latest PR. As for the strings I personally don't see a lot of added value. Devs can use the NotificationManager constants, which can also be found online by people nosing around the stored data.